src/core/server/saved_objects/service/lib/search_dsl/query_params.ts

```diff
@@ -63,25 +63,38 @@ function getFieldsForTypes(types: string[], searchFields?: string[]) {
  */
 function getClauseForType(
   registry: ISavedObjectTypeRegistry,
-  namespace: string | undefined,
+  namespaces: string[] | undefined = ['default'],
   type: string
 ) {
   if (registry.isMultiNamespace(type)) {
     return {
       bool: {
-        must: [{ term: { type } }, { term: { namespaces: namespace ?? 'default' } }],
+        must: [{ term: { type } }, { terms: { namespaces } }],
         must_not: [{ exists: { field: 'namespace' } }],
       },
     };
-  } else if (namespace && registry.isSingleNamespace(type)) {
+  } else if (registry.isSingleNamespace(type)) {
+    const should: Array<Record<string, any>> = [];
+    const eligibleNamespaces = namespaces.filter((namespace) => namespace !== 'default');
+    if (eligibleNamespaces.length > 0) {
+      should.push({ terms: { namespace: eligibleNamespaces } });
+    }
+    if (namespaces.includes('default') ?? true) {
+      should.push({ bool: { must_not: [{ exists: { field: 'namespace' } }] } });
+    }
+    if (should.length === 0) {
+      throw new Error('unhandled search conditions!!');
+    }
```

Comment:
I think this is only possible if `namespaces == []`. Not sure how an Elasticsearch query responds to `terms: {namespaces: []}` but maybe this is invalid input for single and multi-namespace types? Seems impossible for that to be passed down but it makes sense to be defensive. 

Can we make the error message more descriptive like "cannot specify empty namespaces array"

Reply:
Yeah, this is simply a defensive check. I'll update the error text